devito/types/grid.py

```diff
@@ -907,6 +908,180 @@ def bounds(self):
         return self._local_bounds
 
 
+class Border(SubDomainSet):
+    """
+    A convenience class for constructing a SubDomainSet which covers specified edges
+    of the domain to a thickness of `border`. Note that none of the subdomains in this
+    MultiSubDomain will overlap with one another.
```

Comment:
That's not what people usually want, most ABC method requires the domain to overlap in the corner so you can combine the effect of the damping/pml for the different directions. So Ideally this should be an option that default to "overlap"

Reply:
As a single `SubDomainSet`? Or would having N subdomainsets where N is the number of dimensions be better?